tests/lib/tools/suite/tests.backup/task.yaml

```
@@ -0,0 +1,52 @@
+summary: tests for the tests.backup tool
+
+prepare: |
+    mv "${PWD}.tar" "${PWD}.tar.back"
+
+restore: |
+    mv "${PWD}.tar.back" "${PWD}.tar"
+
+execute: |
+    # Without any arguments a help message is printed.
+    tests.backup | MATCH "usage: tests.backup prepare"
+    tests.backup | MATCH "       tests.backup restore"
+
+    # Both -h and --help are also recognized.
+    tests.backup --help | MATCH "usage: tests.backup"
+    tests.backup -h | MATCH "usage: tests.backup"
+
+    # Unknown commands and options are reported
+    tests.backup --foo 2>&1 | MATCH "tests.backup: unknown option --foo"
+    tests.backup foo 2>&1 | MATCH "tests.backup: unknown command foo"
+
+    # Create a file and a directory inside the current path
+    touch "${PWD}/testfile-old"
+    mkdir "${PWD}/testdir-old"
+
+    # Check prepare creates a backup for the current directory
+    [ ! -f "${PWD}.tar" ]
```

Please use `test ...` for assertion-like tests and `[ ... ]` for `if-then-else` statements.
```suggestion
    test ! -f "${PWD}.tar"
```

---

done